src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,767 @@
+                if (!tmpBond.getFlag(CDKConstants.VISITED)) {
+                    if (tmpBond.getFlag(CDKConstants.ISPLACED)) {
+                        if (tmpBond.getBegin().getFlag(CDKConstants.ISPLACED) && tmpBond.getEnd().getFlag(CDKConstants.ISPLACED)) {
+                            tmpChainFragmentationContainer.addBond(tmpBond);
+                            tmpBond.setFlag(CDKConstants.VISITED, true);
+                        }
+                    }
+                    else if (!tmpBond.getFlag(CDKConstants.ISPLACED)) {
+                        tmpRingFragmentationContainer.addBond(tmpBond);
+                        tmpBond.setFlag(CDKConstants.VISITED, true);
+                    }
+                }
+            }
+            //connectivity checking
+            try {
+                if (!tmpRingFragmentationContainer.isEmpty()) {
+                    if (!ConnectivityChecker.isConnected(tmpRingFragmentationContainer)) {
+                        IAtomContainerSet tmpContainerSet = ConnectivityChecker.partitionIntoMolecules(tmpRingFragmentationContainer);
+                        for (IAtomContainer tmpContainer: tmpContainerSet.atomContainers()) {
+                            tmpFragments.addAtomContainer(tmpContainer);
+                        }
+                    } else {
+                        tmpFragments.addAtomContainer(tmpRingFragmentationContainer);
+                    }
+                }
+                if (!tmpChainFragmentationContainer.isEmpty()) {
+                    if (!ConnectivityChecker.isConnected(tmpChainFragmentationContainer)) {
+                        IAtomContainerSet tmpContainerSet = ConnectivityChecker.partitionIntoMolecules(tmpChainFragmentationContainer);
+                        for (IAtomContainer tmpContainer : tmpContainerSet.atomContainers()) {
+                            tmpFragments.addAtomContainer(tmpContainer);
+                        }
+                    } else {
+                        tmpFragments.addAtomContainer(tmpChainFragmentationContainer);
+                    }
+                }
+            } catch (Exception e) {
+                AlkylStructureFragmenter.this.logger.log(Level.WARNING, e + " Connectivity Checking failed");
+                throw new IllegalArgumentException("An Error occurred during Connectivity Checking: " + e.toString() +
+                        " Molecule Name: " + aMolecule.getProperty(Importer.MOLECULE_NAME_PROPERTY_KEY));
+            }
+        } catch (Exception e) {
+            AlkylStructureFragmenter.this.logger.log(Level.WARNING, e + " Fragment Extraction failed");
+            throw new RuntimeException(e);
+        }
+        //</editor-fold>
+        //
+        //<editor-fold desc="Hydrogen Saturation" defaultstate="collapsed">
+        List<IAtomContainer> tmpProcessedFragments = new ArrayList<>(tmpFragments.getAtomContainerCount());
+        try {
+            if (!tmpFragments.isEmpty() && !(tmpFragments == null)) {
```

Comment:
Condition '!(tmpFragments == null)' is always 'true' when reached.
And as Jonas mentioned above please use != instead of !(... == ...)